Solution/Javier Flores

jffc-dev opened the pull request:
Solving Module 3 (Asynchronous Programming) challenges

jffc-dev committed:
feat: 1-callback challenge solved

jffc-dev committed:
feat: reading data from the terminal with process.argv

jffc-dev committed:
feat: util.promisified applied to validate function

jffc-dev committed:
feat: inital solution of challenge 2-promises

jffc-dev committed:
feat: challenge 2-promises solved

jffc-dev committed:
feat(3-async/await): challenge solved using all and allSettled methods

jffc-dev committed:
feat(3-async/await): challenge improved using race and any Promise methods

bryanavalosdev commented on 3-async-await/solution.js:
```diff
@@ -20,16 +23,151 @@ Example:
-    // You generate your id value here
+    } catch (error) {
+        throw new CustomError(error.message, 'price', id)
+    } finally {
```
avoid using empty finally

bryanavalosdev commented on 3-async-await/solution.js:
```diff
@@ -20,16 +23,151 @@ Example:
-    // You generate your id value here
+    } catch (error) {
+        throw new CustomError(error.message, 'price', id)
+    } finally {
```
Avoid using empty finally

jffc-dev replied:
Indeed, that was an unnecesary finally block, I've mremoved it in this commit: https://github.com/ravnhq/JS-async-Nerdery-Challenges/pull/33/commits/058850dcf79252d35285d183d38cf05091aab519

bryanavalosdev commented on 3-async-await/solution.js:
```diff
@@ -20,16 +23,151 @@ Example:
 
-    // You generate your id value here
+    } catch (error) {
+        throw new CustomError(error.message, 'price', id)
```
Improve how you are tracking the errors, 'price' is harcoded as the origin here

jffc-dev replied:
You're right, I accidentally forgot to change it. It has been resolved in this commit: https://github.com/ravnhq/JS-async-Nerdery-Challenges/pull/33/commits/058850dcf79252d35285d183d38cf05091aab519

bryanavalosdev commented on 3-async-await/solution.js:
```diff
@@ -20,16 +23,151 @@ Example:
+        const {value:price} = priceResponse
 
-    // Log the results, or errors, here
+        const product = new Product(id, name, price)
+        return product
```
You can return directly the instance of the produc

jffc-dev replied:
You're right. It has been resolved in this commit: https://github.com/ravnhq/JS-async-Nerdery-Challenges/pull/33/commits/058850dcf79252d35285d183d38cf05091aab519

bryanavalosdev commented on 3-async-await/solution.js:
```diff
@@ -20,16 +23,151 @@ Example:
+
+const generateId = async(delay) => {
+    if(delay){
+        await setTimeout(100)
```
You can use the delay parameter

jffc-dev replied:
That was the idea originally, but I either forgot or accidentally changed it during some testing. It has already been resolved in this commit: https://github.com/ravnhq/JS-async-Nerdery-Challenges/pull/33/commits/058850dcf79252d35285d183d38cf05091aab519

jffc-dev committed:
refactor(3-async/await): apply feedback improvements from code review

- Removed empty `finally` block to improve code clarity
- Refactored error tracking to avoid hardcoding 'price' as the origin
- Simplified return by directly returning the `Product` instance
- Utilized `delay` parameter in `generateId` function for better functionality

jffc-dev commented:
https://github.com/ravnhq/JS-async-Nerdery-Challenges/pull/33/commits/058850dcf79252d35285d183d38cf05091aab519
test
